lib/ansible/modules/web_infrastructure/ansible_tower/tower_project.py

```diff
@@ -147,14 +147,15 @@ def main():
                 except (exc.NotFound) as excinfo:
                     module.fail_json(msg='Failed to update project, organization not found: {0}'.format(organization), changed=False)
 
-                if scm_type:
+                credential = None
+                if scm_credential:
                     try:
                         cred_res = tower_cli.get_resource('credential')
                         cred = cred_res.get(name=scm_credential)
                     except (exc.NotFound) as excinfo:
                         module.fail_json(msg='Failed to update project, credential not found: {0}'.format(scm_credential), changed=False)
+                    credential = cred['id']
```

Comment:
I'd rather not have a variable `scm_credential` and `credential`. You could just call this `scm_credential`, and supply that in the `project.modify` call.